sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala

```diff
@@ -649,17 +658,20 @@ abstract class UpgradesSpec(val suffix: String)
       }
 
       if (validateV1Checked) {
-        cantonLogSrc should include(
+        filterLog(cantonLogSrc, testPackageV2Id) should include(
           s"Package $testPackageV2Id claims to upgrade package id $testPackageV1Id"
         )
       }
 
       failureMessage match {
         // If a failure message is expected, look for it in the canton logs
-        case Some(additionalInfo) => {
-          cantonLogSrc should include(
-            s"The DAR contains a package which claims to upgrade another package, but basic checks indicate the package is not a valid upgrade err-context:{additionalInfo=$additionalInfo"
-          )
+        case Some(additionalInfo) =>
+          if (
+            !cantonLogSrc.contains(
```

Comment:
The err-context contains the package id as well, I think - you could use `filterLog` here as well, rather than having to have a custom failure message.

Reply:
If the error have nothing to do with upgrade checking, I will likely get a pretty confusing empty error.
I think in case of failure it is better for the dev to look directly at the canton log.